### PR TITLE
[FEATURE] Provide migration for extbase magic methods

### PR DESCRIPTION
**Pull request by helsner**
Resolves: #3600

### CODE REVIEW

**Comment by helsner** [comment]
i could need some help her @simonschaufi 
I marked TODOs to check

File: src/Rector/v12/v3/typo3/MigrateMagicRepositoryMethodsRector.php [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    {
+        return [MethodCall::class];
+    }
+
+    /**
+     * @param Node\Expr\MethodCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        // TODO: this doesn't work with my ExampleRepo Stub
```
Comment (simonschaufi) [inline src/Rector/v12/v3/typo3/MigrateMagicRepositoryMethodsRector.php:36]: no problem for me. I never return null here.

File: src/Rector/v12/v3/typo3/MigrateMagicRepositoryMethodsRector.php [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        if (! $this->nodeTypeResolver->isMethodStaticCallOrClassMethodObjectType(
+            $node,
+            new ObjectType('TYPO3\CMS\Extbase\Persistence\Repository')
+        )) {
+            return null;
+        }
+
+        // TODO: check if method already exists in class
+
+        $methodReflection = $this->reflectionResolver->resolveMethodReflectionFromMethodCall($node);
```
Comment (simonschaufi) [inline src/Rector/v12/v3/typo3/MigrateMagicRepositoryMethodsRector.php:56]: I hope this helps you further. I found some nice code here: `\Rector\Core\NodeAnalyzer\DoctrineEntityAnalyzer::hasClassReflectionAttribute`

File: src/Rector/v12/v3/typo3/MigrateMagicRepositoryMethodsRector.php (hunk 1 above)
Comment (helsner) [reply src/Rector/v12/v3/typo3/MigrateMagicRepositoryMethodsRector.php:45]: I created a test stub class that extends the Repository class
With the stub class the check here does not work which is a problem i guess

**Pull request merged**